api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirEncounterServiceImplTest.java

```diff
@@ -70,16 +104,84 @@ public void shouldGetEncounterByUuid() {
 	}
 	
 	@Test
-	public void findEncountersByPatientIdentifier_shouldReturnCollectionOfEncounters() {
-		when(dao.findEncountersByPatientIdentifier(PATIENT_IDENTIFIER))
-		        .thenReturn(Collections.singletonList(openMrsEncounter));
+	public void searchForEncounter_shouldReturnCollectionOfEncounterByDate() {
+		Collection<Encounter> encounters = new ArrayList<>();
```

Comment:
Can these statements be ordered in a more logical fashion? This is very confusing to read.